runtime.go

```diff
@@ -777,3 +781,36 @@ func ToGoHTML(ctx context.Context, c Component) (s template.HTML, err error) {
 	s = template.HTML(b.String())
 	return
 }
+
+// WriteExtractedString is used when rendering templates in development mode.
+// the generator would have written non-go code to the _templ.txt file, which
+// is then read by this function and written to the output.
+func WriteExtractedString(w *bytes.Buffer, lineNum int) error {
```

Comment:
I understand why the name is this, but in my mind, it makes sense to be called `GetWatchModeString` or something.

I think that this should be optimised further to maintain a global variable for last modified date on the strings file, and to load the whole file into RAM, so we don't hit the disk as often, since with this design it looks like each string literal would result in a filesystem read.

So I think... split the file into lines, and keep the array of strings. Reload the array of strings if the file has changed since we last read it.

Reply:
I renamed it, but to `WriteWatchModeString` instead since it doesn't return the string but writes it.

I will think of a way to maintain the state so that we don't hit the filesystem as often. We would need to maintain a struct for each file and somehow initialize it at the start.

Reply:
Found a way to do the caching using a global map.

I decided to leave out having a mutex since I don't think there is much danger here. The worst case is that multiple calls overwrite the same entry in the map if they're fast enough, but I don't think that's such a big problem.

Reply:
I was thinking of something like this...

```go
package templ

var watchState map[string]*WatchState
var watchStateMutex sync.Mutex = new(sync.Mutex)

type WatchState struct {
  LastMod time.Time
  Strings []string
}

func WriteWatchModeString(fileName string, index int, w io.Writer) error {
  watchStateMutex.Lock()
  defer watchStateMutex.Unlock()

  state, ok := watchState[fileName]
  if !ok {
    //TODO: Load the state and store it in the map.
    state = new(WatchState)
    //TODO: Populate the state by reading from disk.
    watchState[fileName] = state
  }
  //TODO: Re-os.Stat the file if the last mod is > 100ms in the past or something (basically, don't thrash the disk).
  //TODO: Re-read the file if it has been modified and update the state.
  //TODO: Return the string. 
} 
```

Reply:
Ah, I see you beat me to it! I'll read through the updates tomorrow!

Reply:
Cleaned it up a bit more, introduced the 100ms buffer and a mutex